### ISSUE TITLE
Dataframe appears empty on mobile

### ISSUE THREAD

**Issue by abidlabs** [gradio-app/gradio#7349]
### Describe the bug

Dataframe appears empty on mobile

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

https://chat.lmsys.org/?leaderboard

(working on better repro)

### Screenshot

![image](https://github.com/gradio-app/gradio/assets/1778297/9d761ff0-c966-4f5d-a831-4b593256ddf4)

### System Info

```shell
gradio==4.17.0
```

### Severity

Blocking usage of gradio

**Comment by pngwn** [comment]
Okay, i have managed to create a minimal repro for this here: https://pngwn-7349-df-repro.hf.space/ 

It is only reproducible on specific devices. Here is a browserstack test with iPhone 15 / Safari 17.2 / Chrome.

https://github.com/gradio-app/gradio/assets/12937446/b17a195c-1bf2-4d76-98d7-6121a7479c1a

This bug occurs when the Dataframe is in a hidden tab. Tabs only set the display properties of the contents they contain, they don't remove things from the page. I believe that the dataframe is trying to measure things when it is mounted to the DOM, which in this case is when it is invisible. I don't think it _re_measures itself when the visibility changes.

I do not know why iOS has different behaviour in chrome/ safari because they have the same rendering engine. Weird but it is what it is.

Things that don't affect it (as far as I can tell):

- The size of the dataframe, the size of the contents, the size of the window (and any strange combination of those, such as the DF being larger than the window, the DF contents being larger than the DF, etc.)
- The data types. I wondered if markdown was causing this but it isn't. It always happens.
- The amount of content (large datasets), i thought this issue was due to leadboard datasets being generally quite large (many columns/ rows) but the main lmsys leaderboard isn't even that big and as you can see in the repro, it doesn't even matter.
- Slow devices. I have testsed with very large datasets with CPU slow down and network slow down and the behaviour is the same. I cant repro it locally.

In short this happens very very consistently as long as the Dataframe is first rendered in a hidden tab.

I have not yet worked out how to fix it but I'm working on that now.
